registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java

```diff
@@ -599,6 +618,80 @@ public String getFilename() {
 
 		return response.getDataShare().getUrl();
 	}
+	private void validateBiometricRecord(BiometricRecord biometricRecord, List<String> modalities, Map<String, String> metaInfoMap) throws DataShareException, JsonParseException, JsonMappingException, IOException {
+		
+		if(modalities==null || modalities.isEmpty()) {
+			throw new DataShareException("Data Share Policy Modalities were Empty");
+
+		}
+		if(biometricRecord == null  || biometricRecord.getSegments() == null
+				|| biometricRecord.getSegments().isEmpty()) {
+			throw new DataShareException("No Biometrics Found with Data Share Policy");
+
+		}	
+		
+		String exceptionBometricsMap = metaInfoMap.get("exceptionBiometrics");
+		
+
+		 ObjectMapper mapper = new ObjectMapper();
+		Map<String, Map<String,Object>> metaInfoExceptionBiometrics = mapper.readValue(exceptionBometricsMap, new TypeReference<Map<String, Map<String,Object>>>() {
+		        });
+		
+		
+		Set<String> exceptionList =  metaInfoExceptionBiometrics.get("applicant").keySet();
+		
+
+		boolean isBioFound = false;
+		for(String modality : modalities) {
+			
+			if(biometricModalitySegmentsMap.containsKey(modality)) {
+				
+				for(String segment : biometricModalitySegmentsMap.get(modality)) {
+					
+					Optional<BIR> optionalBIR = null;
+					if(segment.equalsIgnoreCase("Face")) {
+						
+						optionalBIR = biometricRecord.getSegments().stream()
+								.filter(bir -> bir.getBdbInfo().getType().get(0).equals(BiometricType.FACE)).findFirst();
+					} else {
+						
+						String[] segmentArray = segment.split(" ");
+					
+						optionalBIR = biometricRecord.getSegments().stream()
+								.filter(bir -> bir.getBdbInfo().getSubtype().size() == segmentArray.length ?

```

Comment:
if length is same, length cannot be 2 ? Why are we checking for get(0) alone ?

Reply:
Length Can be 2 Ex: Right MiddleFinger , 0th index value will be "Right" and 1st index value will be "MiddleFinger"

Reply:
ok got it, since you are checking only for get(1) can you check the condition of equal to 2 rather than greater than 1 ?

Reply:
Modified condition for equals 2